cloudscape-design/components
# fix: Print dev warning when the value of Table stickyHeader changes

### Description

`stickyHeader` should stay constant during the lifecycle of the Table. Therefore we:

- print a dev warning if it changes
- document this requirement in the API docs

A similar case we have in the system from which I loosely took as a reference is the `dateOnly` prop of the Date Range Picker: see [API docs](https://github.com/cloudscape-design/components/blob/main/src/date-range-picker/interfaces.ts#L55), [dev warning code](https://github.com/cloudscape-design/components/blob/main/src/date-range-picker/index.tsx#L223).

Ticket: AWSUI-40380

### How has this been tested?

- Added a unit test
- Manually tested on a dev page
- Run a dry run to make sure that console warnings do not break any package build

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
